operator/pkg/ciliumendpointslice/controller.go

```diff
@@ -103,7 +104,11 @@ func registerController(p params) error {
 	if err != nil {
 		return err
 	}
-	if !clientset.IsEnabled() || !p.SharedCfg.EnableCiliumEndpointSlice {
+	if cmp.Or(
+		!clientset.IsEnabled(),
+		!p.SharedCfg.EnableCiliumEndpointSlice,
+		p.SharedCfg.DisableNetworkPolicy,
```

Comment:
Why does this need two "knobs"? What exactly is the case where `p.SharedCfg.EnableCiliumEndpointSlice` is true, but the CiliumEndpointSlice controller should not run?

Reply:
Mainly because we want to follow the pattern of "Only run the processes (controllers) that are being used based on configuration". It doesn't need to run if network policies are disabled. I know this has negligible impact on scalability since cilium-operator is a single leader pod. It still makes sense to follow the pattern.

Additionally, I want to always have CES enabled in my config, so I don't have to change it depending on network policy enablement.

Reply:
right, but this is an "or", which means that `DisableNetworkPolicy` overrides `EnableCiliumEndpointSlice`, which is counterintuitive. Other things besides policy rely on CEP / CES.

Reply:
Network policies are disabled only when CEP CRD is disabled, so CEP / CES won't be used anyway.

I'm not sure what's best to do here. We could add CEPDisabled flag to this "or" condition, which is part of network policy condition. Or we can completely remove the condition.

Reply:
@squeed, do you have a suggestion?

My opinion is that it's acceptable to keep the "or" `DisableNetworkPolicy` condition, since it already contains the condition of disabling CEP (https://github.com/cilium/cilium/blob/main/pkg/option/features.go#L16), which should automatically affect CES usage.